java-checks/src/main/java/org/sonar/java/checks/InterruptedExceptionCheck.java

```diff
@@ -64,17 +70,31 @@ public void visitNode(Tree tree) {
     TryStatementTree tryStatementTree = (TryStatementTree) tree;
     withinInterruptingFinally.addFirst(isFinallyInterrupting(tryStatementTree.finallyBlock()));
     for (CatchTree catchTree : tryStatementTree.catches()) {
-      Type catchType = catchTree.parameter().symbol().type();
-      if(catchType.is("java.lang.InterruptedException") || catchType.is("java.lang.ThreadDeath")) {
+      Optional<Type> interruptType = findInterruptingType(catchTree.parameter());
+      if(interruptType.isPresent()) {
         BlockVisitor blockVisitor = new BlockVisitor(catchTree.parameter().symbol());
         catchTree.block().accept(blockVisitor);
         if(!blockVisitor.threadInterrupted && !isWithinInterruptingFinally()) {
-          reportIssue(catchTree.parameter(), "Either re-interrupt this method or rethrow the \""+catchType.name()+"\".");
+          reportIssue(catchTree.parameter(), "Either re-interrupt this method" +
+            " or rethrow the \""+interruptType.get().name()+"\".");
         }
       }
     }
   }
 
+  private static Optional<Type> findInterruptingType(VariableTree parameter) {
```

Comment:
No need to use optional here, just use anyMatch and return a boolean

Reply:
The method return the type and not a boolean to be able to add the type name in the issue message (line 79).